lib/azure/armrest/storage_account_service.rb

```diff
@@ -108,33 +107,85 @@ def create(account_name, rgroup = configuration.resource_group, options)
         acct
       end
 
-      # Returns the primary and secondary access keys for the given
-      # storage account as a hash.
+      # Returns the primary and secondary access keys for the given storage
+      # account. This method will return a hash with 'key1' and 'key2' as its
+      # keys.
+      #
+      # If you want a list of StorageAccountKey objects, then use the
+      # list_account_keys2 method instead.
       #
       def list_account_keys(account_name, group = configuration.resource_group)
         validate_resource_group(group)
 
         url = build_url(group, account_name, 'listKeys')
         response = rest_post(url)
-        JSON.parse(response)
+        hash = JSON.parse(response.body)
+
+        parse_account_keys_from_hash(hash)
+      end
+
+      alias list_storage_account_keys list_account_keys
+
+      # Returns a list of StorageAccountKey objects consisting of information
+      # the primary and secondary keys. This method requires an api-version
+      # string of 2016-01-01 or later, or an error is raised.
+      #
+      # If you want a plain hash, use the list_account_keys method instead.
+      #
+      def list_account_key_objects(account_name, group = configuration.resource_group)
+        validate_resource_group(group)
+
+        unless Time.parse(api_version).utc >= Time.parse('2016-01-01').utc
+          raise ArgumentError, "unsupported api-version string '#{api_version}'"
+        end
+
+        url = build_url(group, account_name, 'listKeys')
+        response = rest_post(url)
+        JSON.parse(response.body)['keys'].map { |hash| StorageAccountKey.new(hash) }
+      end
+
+      alias list_storage_account_key_objects list_account_key_objects
+
+      # Regenerates the primary or secondary access keys for the given storage
+      # account. The +key_name+ may be either 'key1' or 'key2'. If no key name
+      # is provided, then it defaults to 'key1'.
+      #
+      def regenerate_account_keys(account_name, group = configuration.resource_group, key_name = 'key1')
+        validate_resource_group(group)
+
+        options = {'keyName' => key_name}
+
+        url = build_url(group, account_name, 'regenerateKey')
+        response = rest_post(url, options.to_json)
+        hash = JSON.parse(response.body)
+
+        parse_account_keys_from_hash(hash)
       end
```

Comment:
@bronaghs The regenerate_account_keys method? I'm not, it's a public facing method that users can call.